### DISCUSSION TITLE
how to automatically display the virtual space corresponding to the desktop when there are multiple monitors

### DISCUSSION THREAD

**Discussion by ati46** [FelixKratz/SketchyBar#128] [Q&A] [answered]
There are ten virtual desktops by default. I hope that all virtual desktop menus can be displayed on the main desktop when the external monitor is not connected. When an external monitor is connected, the respective virtual desktop menus can be displayed on different monitors instead of the main desktop. All menus are displayed on the screen.
I used Übersicht to implement it before, but Übersicht sometimes occupies too much resources. Later I found SketchyBar. I like it very much but I can’t implement the above functions. Can you help me?

**Comment by FelixKratz** [comment]
Are you using yabai? This could nicely be accomplished with a small script that querys yabai for the spaces. I will see if I get some time to implement something like this myself since it seems to be a popular request. This could be a good starting point for such a script: [https://github.com/FelixKratz/SketchyBar/discussions/12#discussioncomment-1286504](https://github.com/FelixKratz/SketchyBar/discussions/12#discussioncomment-1286504)

(I converted the issue into a discussion such that others can find it more easily later on)

**Comment by ati46** [comment]
@FelixKratz yes it solved my problem very well, thank you very much

**Comment by FelixKratz** [comment, marked as answer]
In the new version this is very easy to achieve, see for example the default config: https://github.com/FelixKratz/SketchyBar/blob/master/sketchybarrc
all you have to do is, not specifying an `associated_display` for the space.

**Reply by diegolhambi** [reply]
How I can add new spaces created after the initialization of the bar?
Even when the space item associated with the new space exists in the bar it only show when I change to the recent created space.

UPDATE: 
Sending in terminal `sketchybar --update` force show the recent created space that has item.

**Reply by FelixKratz** [reply]
You can add as many spaces as you like in the beginning, if the space is not available and you did not associate a display it will not be shown. Then when you create a new space and switch spaces it is shown.
When you create new spaces you have to give them their initial properties with a --set command, otherwise they are indeed not shown as the `script` is only executed on space *and* selection change.

Running —update from within an item script can potentially generate an endless loop, —trigger space_change should also not be called from within the script of space components for exactly the same reason. I will handle this case in the next update more elegantly (#208).
